cinema/views.py

```diff
@@ -45,9 +69,20 @@ def get_serializer_class(self):
 
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
-    queryset = MovieSession.objects.all()
+
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = MovieSession.objects.select_related("movie")
+        date = self.request.query_params.get("date")
+        movie_id = self.request.query_params.get("movie")
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+        if movie_id:
+            # movie = [int(str_id) for str_id in movie.split(",")]
```

Comment:
There is a commented-out line intended to convert `movie` query parameter to a list of integers. If this functionality is not needed, consider removing the comment to keep the code clean. If it is needed, ensure it is correctly implemented and integrated.